keps/sig-network/0031-20181109-multicast-support.md

```diff
@@ -0,0 +1,191 @@
+---
+kep-number: 31
+title: Multicast Support
+authors:
+  - "@danwinship"
+owning-sig: sig-network
+reviewers:
+  - TBD
+approvers:
+  - "@thockin"
+editor: TBD
+creation-date: 2018-11-09
+last-updated: 2018-11-09
+status: provisional
+---
+
+# Multicast Support
+
+## Table of Contents
+
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+* [Existing Work](#existing-work)
+* [Proposal](#proposal)
+    * [User Stories](#user-stories)
+    * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+    * [Risks and Mitigations](#risks-and-mitigations)
+* [Graduation Criteria](#graduation-criteria)
+* [Implementation History](#implementation-history)
+
+## Summary
+
+Currently, support for IP multicast traffic on the pod network is
+entirely undefined, with no multicast-related behavior being either
+required or forbidden, and different plugins offering different
+functionality. This KEP aims to standardize things.
+
+## Motivation
+
+Some Kubernetes users have workloads that require multicast traffic,
+and need to be able to confirm that multicast traffic will flow
+between the desired endpoints, but not be visible to other endpoints.
+
+Other Kubernetes users are more concerned with simply ensuring that
+undesired multicast traffic is blocked, to prevent a single pod from
+being able to easily saturate the network.
+
+See also:
```

Comment:
One of the things I'd like to see a bit more of in this KEP is a high-level justification for what multicast configuration knobs the KEP is proposing to add to Kubernetes itself, and a justification for why those things need to live in Kubernetes.

For example, it seems to me that Weave Net and OpenShift both support multicast to some extent today just fine, so what's the reason to now add something to Kubernetes?